app/transport/http/contracts.go

```diff
@@ -6,5 +6,5 @@ import (
 
 // UserUsecase represents User use-case layer
 type UserUsecase interface {
-	Signup(entities.User) error
+	Signup(*entities.User) error
```

Comment:
Do you really need to pass a pointer to User?

Reply:
@vpakhuchyi  I will try to explain. In the previous version of the code, we simply added the user to the database and at the same time logged all his fields, but the ID was empty because the ID is created in the database. And it seemed normal. Now we have decided that almost all user fields are sensitive data except ID (which is empty). There is no point in logging an empty value. So I changed the code so that the user ID is returned from the database after adding

Reply:
fixed